packages/datepickers/src/elements/DatepickerRange/components/End.tsx

```diff
@@ -30,31 +34,49 @@ export const End = (props: PropsWithChildren<HTMLAttributes<HTMLInputElement>>)
     [dispatch, props.children]
   );
 
+  const handleBlur = useCallback(() => {
+    dispatch({ type: 'END_BLUR' });
+    let parsedDate;
+
+    if (customParseDate) {
+      parsedDate = customParseDate(state.endInputValue);
+    } else {
+      parsedDate = parseInputValue({
+        inputValue: state.endInputValue
+      });
+    }
+
+    if (onChange && parsedDate && isValid(parsedDate) && !isSameDay(parsedDate, endValue!)) {
+      onChange && onChange({ startValue, endValue: parsedDate });
+    }
+  }, [dispatch, onChange, startValue, endValue, customParseDate, state.endInputValue]);
+
   const onKeydownCallback = useCallback(
     (e: React.KeyboardEvent<HTMLInputElement>) => {
       if (e.keyCode === KEY_CODES.ENTER) {
-        dispatch({ type: 'END_BLUR' });
+        handleBlur();
+
         e.preventDefault();
       }
 
       (props.children as any).props.onKeyDown && (props.children as any).props.onKeyDown(e);
     },
-    [dispatch, props.children]
+    [handleBlur, props.children]
   );
 
   const onBlurCallback = useCallback(
     (e: React.FocusEvent<HTMLInputElement>) => {
-      dispatch({ type: 'END_BLUR' });
+      handleBlur();
 
       (props.children as any).props.onBlur && (props.children as any).props.onBlur(e);
     },
-    [dispatch, props.children]
+    [handleBlur, props.children]
   );
 
   const childElement = React.Children.only(props.children as React.ReactElement);
 
   return React.cloneElement(childElement, {
-    value: state.endInputValue,
+    value: state.endInputValue || '',
```

Comment:
Prevents switching React input from a controlled component to an uncontrolled component. This is a band-aid fix, but necessary given the current architecture of `DatepickerRange`.